### PR TITLE
Implement RFC-0096 Remove Stacks & Mixins: phase 1

### PR DESCRIPTION
**Pull request by jkutner**
## Summary

Phase of [RFC #0096](https://github.com/buildpacks/rfcs/blob/main/text/0096-remove-stacks-mixins.md) - Remove Stacks & Mixins implementation.

#### Before

Support `[stack]` in `builder.toml, and write `/cnb/stack.toml` in builder.

#### After

Support both `[stack]` and `[run]`/`[build]` tables. If both are present, `[run]`/`[build]` take precedent. Stack ID will still be accepted. 

A `/cnb/stack.toml` will be added to the builder if a Stack ID is provided. A `/cnb/run.toml` may also be added to the builder if `[run]`/`[build]` are definied.

## Documentation

- Should this change be documented?
    - [x] Yes, see #___
    - [ ] No

## Related

Resolves #___

### CODE REVIEW

File: internal/builder/builder_test.go [hunk 1]
```diff
@@ -240,15 +240,15 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
...
 				it.Before(func() {
 					h.AssertNil(t, baseImage.SetEnv("CNB_USER_ID", "1234"))
 					h.AssertNil(t, baseImage.SetEnv("CNB_GROUP_ID", "4321"))
 				})
 
-				it("returns an error", func() {
+				it("does not return an error", func() {
 					_, err := builder.New(baseImage, "some/builder")
-					h.AssertError(t, err, "image 'base/image' missing label 'io.buildpacks.stack.id'")
+					h.AssertNilE(t, err)
```
Comment (natalieparellano) [inline internal/builder/builder_test.go:251]: Why don't we error here?

File: internal/build/container_ops.go [hunk 2]
```diff
@@ -228,6 +228,37 @@ func WriteStackToml(dstPath string, stack builder.StackMetadata, os string) Cont
...
 
+// WriteRunToml writes a `run.toml` based on the RunConfig provided to the destination path.
+func WriteRunToml(dstPath string, runImages []builder.RunImageMetadata, os string) ContainerOperation {
+	return func(ctrClient DockerClient, ctx context.Context, containerID string, stdout, stderr io.Writer) error {
+		buf := &bytes.Buffer{}
+		err := toml.NewEncoder(buf).Encode(builder.RunImages{
+			Images: runImages,
+		})
+		if err != nil {
+			return errors.Wrap(err, "marshaling stack metadata")
```
Comment (natalieparellano) [inline internal/build/container_ops.go:239]:
```suggestion
			return errors.Wrap(err, "marshaling run metadata")
```

**Review by natalieparellano — APPROVED** [review]
Looks good @jkutner!

File: internal/builder/builder_test.go (hunk 1 above)
Comment (natalieparellano) [reply internal/builder/builder_test.go:251]: Ah, makes sense - I guess what threw me was the addition of `and run image` to the test description. I thought of a builder that is missing a run image reference.

**Review by hone — APPROVED** [review]

**Review by dfreilich — APPROVED** [review]

**Pull request merged**